pkgs/development/python-modules/flaxlib/default.nix

```diff
@@ -0,0 +1,42 @@
+{
+  lib,
+  buildPythonPackage,
+  flax,
+  rustPlatform,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "flaxlib";
+  version = "0.0.1a1";
+  pyproject = true;
+
+  inherit (flax) src;
```

Comment:
This is not ideal. `0.0.1a1` is the version reported on Pypi: https://pypi.org/project/flaxlib.
However, I use `inherit (flax) src` to make sure that they remain synced.

Reply:
Indeed because when we update the main one, this one will likely be missed. Can we check in bash code if version matches with the source code?

Reply:
Good idea, done.
Please, don't hesitate to propose modification for my check script.

Reply:
a problem with this approach is that a flax bump will likely break flaxlib, which means r-ryantm will make no PRs for this package

Reply:
You are right. Doing this in the update script is a good idea, however I foresee some problem:
It suffers from a circular dependency. To get the new version of `flaxlib`, you need to inspect the content of the new version of `flax`.
This would suggest updating `flax` first. However, `flaxlib` is itself a dependency of `flax`, so we might not be able to successfully update `flax` first... Apart if we accept that it won't build until we also update `flaxlib`...

Reply:
```
nix-update flax
nix-build . -A flax.src
nix-update flaxlib --version="$(tomlq <result/Cargo.toml .something.version)" --commit
```

might do the trick

Reply:
I would prefer to merge this update as is. I don't have the time currently to look at this.
I will do the updates manually for now.

Reply:
At list it cannot silently fail, which is the most important I guess.